pkg/hubble/peer/types/client.go

```diff
@@ -54,3 +58,32 @@ func (b LocalClientBuilder) Client(target string) (Client, error) {
 	}
 	return &client{conn, peerpb.NewPeerClient(conn)}, nil
 }
+
+// RemoteClientBuilder is a ClientBuilder that is suitable when the gRPC
+// connection to the Peer service is remote (typically a K8s Service).
+type RemoteClientBuilder struct {
+	DialTimeout   time.Duration
+	TLSConfig     certloader.ClientConfigBuilder
+	TLSServerName string
+}
+
+// Client implements ClientBuilder.Client.
+func (b RemoteClientBuilder) Client(target string) (Client, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), b.DialTimeout)
+	defer cancel()
```

Comment:
nit: could we move the ctx creation just before we use it (ie above the `grpc.DialContext()` call)?